d3rlpy/base.py

```diff
@@ -336,15 +340,27 @@ def fit(self,
             if new_transitions:
                 transitions = env_transitions + new_transitions
 
-            indices = np.random.permutation(np.arange(len(transitions)))
-            n_iters = len(transitions) // self.batch_size
-            range_gen = trange(n_iters) if show_progress else range(n_iters)
-            for itr in range_gen:
-                # pick transitions
-                sampled_transitions = []
-                head_index = itr * self.batch_size
-                for index in indices[head_index:head_index + self.batch_size]:
-                    sampled_transitions.append(transitions[index])
+            # shuffle data
+            if shuffle:
+                random.shuffle(transitions)
+
+            # batches generator
+            total_batches = ((len(transitions) + self.batch_size - 1) //
+                             self.batch_size)
+            batches = (transitions[i * self.batch_size:(i + 1) *
+                                   self.batch_size]
+                       for i in range(total_batches))
+
+            # Epoch progress bar
+            tqdm_epoch = tqdm(batches,
```

Comment:
Since the above comment, I'd like you to use this to replace `range_gen` above.

Reply:
ok